cvs server: Diffing .
cvs server: Diffing cups
cvs server: Diffing cups/cupsdconf
cvs server: Diffing ext
cvs server: Diffing filters
cvs server: Diffing lpd
cvs server: Diffing lpdunix
cvs server: Diffing management
Index: management/kmspecialprinterdlg.cpp
===================================================================
RCS file: /home/kde/kdelibs/kdeprint/management/kmspecialprinterdlg.cpp,v
retrieving revision 1.4
diff -u -p -r1.4 kmspecialprinterdlg.cpp
--- kdelibs/kdeprint/management/kmspecialprinterdlg.cpp	2001/06/21 12:02:15	1.4
+++ kdelibs/kdeprint/management/kmspecialprinterdlg.cpp	2002/02/14 08:30:58
@@ -40,7 +40,7 @@ KMSpecialPrinterDlg::KMSpecialPrinterDlg
 	setCaption(i18n("Add special printer"));
 
 	// widget creation
-	QPushButton	*m_ok = new QPushButton(i18n("OK"), this);
+	m_ok = new QPushButton(i18n("OK"), this);
 	QPushButton	*m_cancel = new QPushButton(i18n("Cancel"), this);
 	connect(m_ok,SIGNAL(clicked()),SLOT(accept()));
 	connect(m_cancel,SIGNAL(clicked()),SLOT(reject()));
@@ -48,6 +48,7 @@ KMSpecialPrinterDlg::KMSpecialPrinterDlg
 	QPushButton	*m_browse = new QPushButton(i18n("Browse"), this);
 	connect(m_browse,SIGNAL(clicked()),SLOT(slotBrowse()));
 	m_name = new QLineEdit(this);
+	connect(m_name, SIGNAL(textChanged ( const QString & )),this,SLOT(slotTextChanged(const QString & )));
 	m_description = new QLineEdit(this);
 	m_location = new QLineEdit(this);
 	m_command = new QLineEdit(this);
@@ -109,9 +110,13 @@ KMSpecialPrinterDlg::KMSpecialPrinterDlg
 	l3->addStretch(1);
 	l3->addWidget(m_ok);
 	l3->addWidget(m_cancel);
-
+	m_ok->setEnabled(!m_name->text().isEmpty());
 	// resize dialog
 	resize(400,100);
+}
+void KMSpecialPrinterDlg::slotTextChanged(const QString & text)
+{
+  m_ok->setEnabled(!text.isEmpty());
 }
 
 void KMSpecialPrinterDlg::done(int result)
Index: management/kmspecialprinterdlg.h
===================================================================
RCS file: /home/kde/kdelibs/kdeprint/management/kmspecialprinterdlg.h,v
retrieving revision 1.3
diff -u -p -r1.3 kmspecialprinterdlg.h
--- kdelibs/kdeprint/management/kmspecialprinterdlg.h	2001/06/21 12:02:15	1.3
+++ kdelibs/kdeprint/management/kmspecialprinterdlg.h	2002/02/14 08:30:58
@@ -27,6 +27,7 @@ class QLineEdit;
 class QCheckBox;
 class KIconButton;
 class KEditListBox;
+class QPushButton;
 
 class KMSpecialPrinterDlg : public KDialog
 {
@@ -43,12 +44,13 @@ protected:
 protected slots:
 	virtual void done(int);
 	void slotBrowse();
-
+	void slotTextChanged(const QString & text);
 private:
 	QLineEdit	*m_name, *m_description, *m_location, *m_command, *m_extension;
 	QCheckBox	*m_usefile;
 	KIconButton	*m_icon;
 	KEditListBox	*m_require;
+	QPushButton     *m_ok;
 };
 
 #endif
cvs server: Diffing pics
cvs server: Diffing rlpr
cvs server: Diffing slave
cvs server: Diffing slave/mimetypes
cvs server: Diffing slave/pics
cvs server: Diffing slave/templates
cvs server: Diffing tests
